service/src/main/java/bio/terra/cbas/controllers/RunSetsApiController.java

```diff
@@ -63,13 +73,47 @@ public RunSetsApiController(
 
   @Override
   public ResponseEntity<RunSetStateResponse> postRunSet(RunSetRequest request) {
-    if (!requestIsValid(request, this.cbasApiConfiguration)) {
-      return new ResponseEntity<>(HttpStatus.BAD_REQUEST);
+    Gson gson = new Gson();
+
+    // request validation
+    Optional<ResponseEntity<RunSetStateResponse>> validateRequestResponse =
+        validateRequest(request, this.cbasApiConfiguration.getRunSetsMaximumRecordIds());
+    if (validateRequestResponse.isPresent()) {
+      return validateRequestResponse.get();
     }
 
-    // Store the method
-    UUID methodId = UUID.randomUUID();
+    // Fetch WDS Records and keep track of errors while retrieving records
+    String recordType = request.getWdsRecords().getRecordType();
+    List<String> recordIds = request.getWdsRecords().getRecordIds();
+    recordRecordsInRequest(recordIds.size());
 
+    ArrayList<RecordResponse> recordResponses = new ArrayList<>();
+    HashMap<String, String> recordIdsWithError = new HashMap<>();
+    for (String recordId : recordIds) {
+      try {
+        recordResponses.add(wdsService.getRecord(recordType, recordId));
+      } catch (ApiException e) {
+        log.warn("Record lookup for Record ID {} failed.", recordId, e);
+        ErrorResponse error = gson.fromJson(e.getResponseBody(), ErrorResponse.class);
```

Comment:
What happens if this parse from json throws its own exception?

Reply:
That's a good point. I could wrap this in a try and catch. I am doing this because `wdsService.getRecord` throws an `ErrorResponse` with below fields inside the ApiException. So I thought maybe it might be better to only extract the error message and send it to user. Do you think this is a good idea?
Example of exception thrown by WDS:
```
org.databiosphere.workspacedata.client.ApiException: {"timestamp":"2022-10-20T19:45:01.210+00:00","status":404,"error":"Not Found","message":"Instance does not exist","path":"/15f36863-30a5-4cab-91f7-52be439f1175/records/v0.2/FOO/FOO4"}
```